README
NAME
    Dist::Zilla::Plugin::AutoMetaResources - Automagical MetaResources

VERSION
    version 1.110050

SYNOPSIS
        # dist.ini
        name = Dist-Zilla-Plugin-AutoMetaResources
        ...

        [AutoMetaResources]
        bugtracker.rt      = 1
        repository.github  = user:ajgb
        homepage           = http://search.cpan.org/dist/%{dist}

    or in PluginBundle::Easy

        $self->add_plugins(
            [
                AutoMetaResources => {
                    'bugtracker.rt' => 1,
                    'repository.github' => [{ user => 'ajgb' }],
                    'homepage' => 'http://search.cpan.org/dist/%{dist}',
                }
            ]
        );

    both for Dist::Zilla::Plugin::AutoMetaResources would be equivalent of

        [MetaResources]
        bugtracker.web    = https://rt.cpan.org/Public/Dist/Display.html?Name=Dist-Zilla-Plugin-AutoMetaResources
        bugtracker.mailto = bug-Dist-Zilla-Plugin-AutoMetaResources@rt.cpan.org
        repository.url    = git://github.com/ajgb/dist-zilla-plugin-autometaresources.git
        repository.web    = http://github.com/ajgb/dist-zilla-plugin-autometaresources
        repository.type   = git
        homepage          = http://search.cpan.org/dist/Dist-Zilla-Plugin-AutoMetaResources

DESCRIPTION
    If you find repeating the dist name again in the configuration really
    annoying, this plugin comes to the rescue!

CONFIGURATION
    Most of the known resources requires just boolean flag to be enabled,
    but you are free to pass your own options:

        bugtracker.rt =  dist:Other-Name;lcdist:other-name

    which is transformed to:

        [{
            dist => 'Other-Name',
            lcdist => 'other-name',
        }]

  BUGTRACKER
    RT bugtracker:

        [AutoMetaResources]
        bugtracker.rt = 1

        # same as
        [MetaResources]
        bugtracker.web    = https://rt.cpan.org/Public/Dist/Display.html?Name=%{dist}
        bugtracker.mailto = bug-%{dist}@rt.cpan.org

    GitHub issues:

        [AutoMetaResources]
        bugtracker.github = user:ajgb

        # same as
        [MetaResources]
        bugtracker.web    = http://github.com/ajgb/%{lcdist}/issues

  REPOSITORY
    GitHub repository:

        [AutoMetaResources]
        repository.github = user:ajgb
        # or
        repository.GitHub = user:ajgb

        # same as
        [MetaResources]
        repository.url    = git://github.com/ajgb/%{lcdist}.git
        repository.web    = http://github.com/ajgb/%{lcdist}
        repository.type   = git

    Git Moose repository:

        [AutoMetaResources]
        repository.gitmo = 1

        # same as
        [MetaResources]
        repository.url    = git://git.moose.perl.org/%{dist}.git
        repository.web    = http://git.shadowcat.co.uk/gitweb/gitweb.cgi?p=gitmo/%{dist}.git;a=summary
        repository.type   = git

    Catalyst Subversion repository:

        [AutoMetaResources]
        repository.catsvn = 1

        # same as
        [MetaResources]
        repository.url    = http://dev.catalyst.perl.org/repos/Catalyst/%{dist}/
        repository.web    = http://dev.catalystframework.org/svnweb/Catalyst/browse/%{dist}
        repository.type   = svn

    Shadowcat's Git repositories:

        [AutoMetaResources]
        repository.catagits = 1

        # same as
        [MetaResources]
        repository.url    = git://git.shadowcat.co.uk/catagits/%{dist}.git
        repository.web    = http://git.shadowcat.co.uk/gitweb/gitweb.cgi?p=catagits/%{dist}.git;a=summary
        repository.type   = git

    and

        [AutoMetaResources]
        repository.p5sagit = 1

        # same as
        [MetaResources]
        repository.url    = git://git.shadowcat.co.uk/p5sagit/%{dist}.git
        repository.web    = http://git.shadowcat.co.uk/gitweb/gitweb.cgi?p=p5sagit/%{dist}.git;a=summary
        repository.type   = git

    and

        [AutoMetaResources]
        repository.dbsrgits = 1

        # same as
        [MetaResources]
        repository.url    = git://git.shadowcat.co.uk/dbsrgits/%{dist}.git
        repository.web    = http://git.shadowcat.co.uk/gitweb/gitweb.cgi?p=dbsrgits/%{dist}.git;a=summary
        repository.type   = git

  OTHER
        [AutoMetaResources]
        homepage = http://search.cpan.org/dist/%{dist}

    You are free to pass other valid metadata options to be included as
    resources. Values can contain following variables:

    *%{dist}*
        Name of the distribution

    *%{lcdist}*
        Name of the distribution (lowercase)

AUTHOR
    Alex J. G. Burzyński <ajgb@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Alex J. G. Burzyński
    <ajgb@cpan.org>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for  Dist-Zilla-Plugin-AutoMetaResources

1.20   2013-04-14 02:54:47
    FIXES:
        [rt#84593] Escaping braces in regex for 5.16 compliance.
        Reported and fixed by GARU.

1.110050    Wed Jan  5 20:48:23 2011
            github.com is using HTTPS by default

0.01        Wed Nov  3 23:51:16 2010
            First version.



Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Automagical MetaResources",
  "dist_author" => [
    "Alex J. G. Burzy\305\204ski <ajgb\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-AutoMetaResources",
  "dist_version" => "1.20",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::AutoMetaResources",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla::Role::MetaProvider" => 0,
    "Moose" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "Test::MockObject" => 0,
    "Test::More" => 0,
    "Test::NoWarnings" => 0
  }
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Automagical MetaResources",
  "AUTHOR" => "Alex J. G. Burzy\305\204ski <ajgb\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-AutoMetaResources",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::Plugin::AutoMetaResources",
  "PREREQ_PM" => {
    "Dist::Zilla::Role::MetaProvider" => 0,
    "Moose" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "Test::MockObject" => 0,
    "Test::More" => 0,
    "Test::NoWarnings" => 0
  },
  "VERSION" => "1.20",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



